app/controllers/spree/api/images_controller_decorator.rb

```diff
@@ -0,0 +1,15 @@
+Spree::Api::ImagesController.class_eval do
+  def update_product_image
+    @product = Spree::Product.find(params[:product_id])
+    authorize! :update, @product
+
+    if @product.images.first.nil?
```

Comment:
do we only use an image per product although Spree supports many? I'd like to understand whether those repeated references to `first` actually hide an implicit product decision.

Reply:
If I recall, Spree supports multiple but we don't actually have anything in our views for displaying more than one image in the frontend, so I made this choice to ensure we are only overwriting a single image and not accidentally uploading multiple images whilst not being able to use them.

Reply:
Then, if that's case we should reflect it in the code (a `Product#main_image` perhaps?), rather than having that concept lying implicitly everywhere in the codebase.